app/services/fees/subscription_service.rb

```diff
@@ -84,8 +84,16 @@ def should_compute_upgraded_amount?
     #        or when it is payed in advance on an anniversary base
     def should_use_full_amount?
       return true if plan.pay_in_advance? && subscription.anniversary?
+      return true if subscription.fees.subscription_kind.exists?
 
-      subscription.fees.subscription_kind.exists?
+      return true if subscription.started_in_past? && plan.pay_in_advance?
+
+      if subscription.started_in_past? &&
+        subscription.started_at < date_service(subscription).previous_beginning_of_period
```

Comment:
Layout/MultilineOperationIndentation: Align the operands of a condition in an `if` statement spanning multiple lines.